stl/inc/ranges

```diff
@@ -636,12 +636,138 @@ namespace ranges {
         /* [[no_unique_address]] */ _Ty _Value{};
     };
 
-    // CLASS TEMPLATE ranges::empty_view
+    // CLASS TEMPLATE ranges::_Non_propagating_cache
     // clang-format off
-    template <class _Ty>
-        requires is_object_v<_Ty>
-    class empty_view : public view_interface<empty_view<_Ty>> {
+    template <_Destructible_object _Ty>
+    class _Non_propagating_cache { // a simplified optional that resets on copy / move
+        // clang-format on
+    public:
+        constexpr _Non_propagating_cache() noexcept : _Dummy{} {}
+
+        _CONSTEXPR20 ~_Non_propagating_cache() noexcept {

```

Comment:
As this is in `<ranges>`, I believe that all occurrences of `_CONSTEXPR20` here should be plain `constexpr`. (I suspect that this was derived from `<optional>` originally?)